controller/engine_controller.go

```diff
@@ -578,9 +578,10 @@ func (ec *EngineController) isMonitoring(e *longhorn.Engine) bool {
 }
 
 func (ec *EngineController) startMonitoring(e *longhorn.Engine) {
-	//it's possible for monitor and engineController to send stop signal at
-	//the same time, don't make it block
-	stopCh := make(chan struct{}, 1)
+	// It's possible for monitor and engineController to send stop signal at
+	// the same time, don't make it block:
+	// https://github.com/longhorn/longhorn/issues/1545
+	stopCh := make(chan struct{}, 2)
```

Comment:
Normally I don't think it's a good idea to increase the buffer size of the channel to fix a race condition. What if there is one more signal? May need to think if we can avoid the blocking on the receiving side.